### PR TITLE
tabula-java: build from source

### PR DESCRIPTION
**Pull request by elohmeier**
## Description of changes

This builds tabula-java from source instead of using the binary distribution.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/tabula-java/default.nix [hunk 1]
```diff
@@ -1,24 +1,74 @@
...
-  src = fetchurl {
-    url = "https://github.com/tabulapdf/tabula-java/releases/download/v${version}/tabula-${version}-jar-with-dependencies.jar";
-    sha256 = "sha256-IWHj//ZZOdfOCBJHnPnKNoYNtWl/f8H6ARYe1AkqB0U=";
+  src = fetchFromGitHub {
+    owner = "tabulapdf";
+    repo = "tabula-java";
+    rev = "v${version}";
+    hash = "sha256-lg8/diyGhfkUU0w7PEOlxb1WNpJZVDDllxMMsTIU/Cw=";
   };
+  fetchedMavenDeps = stdenv.mkDerivation ({
```
Comment (wegank) [inline pkgs/applications/misc/tabula-java/default.nix:18]: Can all these be converted to `buildMavenPackage`?

Comment (elohmeier) [reply pkgs/applications/misc/tabula-java/default.nix]: Thanks, done!

**Comment by wegank** [comment]
Probably also needs `-DskipTests` to skip tests that fail on both Linux platforms.

**Comment by elohmeier** [comment]
Thanks, added that!

File: pkgs/applications/misc/tabula-java/default.nix [hunk 2]
```diff
@@ -1,24 +1,37 @@
...
-    sha256 = "sha256-IWHj//ZZOdfOCBJHnPnKNoYNtWl/f8H6ARYe1AkqB0U=";
+  src = fetchFromGitHub {
+    owner = "tabulapdf";
+    repo = "tabula-java";
+    rev = "v${version}";
+    hash = "sha256-lg8/diyGhfkUU0w7PEOlxb1WNpJZVDDllxMMsTIU/Cw=";
   };
 
-  nativeBuildInputs = [ makeWrapper ];
+  mvnHash = "sha256-//JxO7UpU9pxUghfmyUZWhgfxAfy6wHAuwZ6muVk6BE=";
```
Comment (wegank) [inline pkgs/applications/misc/tabula-java/default.nix:19]:
```suggestion
  mvnHash = "sha256-yULCBHgctZZU3Deod+nQujssmUy+kgdFdgE3NUuFhOw=";
```

Comment (elohmeier) [reply pkgs/applications/misc/tabula-java/default.nix]: Thanks, updated!

File: pkgs/applications/misc/tabula-java/default.nix [hunk 3]
```diff
@@ -1,24 +1,37 @@
-{ stdenv, lib, fetchurl, jre, makeWrapper }:
+{ fetchFromGitHub
+, jre
+, lib
+, makeWrapper
+, maven
```
[the remark this answers is not in the document]
Comment (elohmeier) [reply pkgs/applications/misc/tabula-java/default.nix]: Done.

**Pull request merged**